coverity-scan.sh

```diff
@@ -94,54 +110,45 @@ scanit() {
   [ -f netdata-coverity-analysis.tgz ] && run rm netdata-coverity-analysis.tgz
 
   progress "Configuring netdata source..."
+
   run autoreconf -ivf
-  run ./configure --disable-lto \
-    --enable-https \
-    --enable-jsonc \
-    --enable-plugin-nfacct \
-    --enable-plugin-freeipmi \
-    --enable-plugin-cups \
-    --enable-backend-prometheus-remote-write \
-    ${NULL}
-
-  # TODO: enable these plugins too
-  #	--enable-plugin-xenstat \
-  #	--enable-backend-kinesis \
-  #	--enable-backend-mongodb \
+  run ./configure ${OTHER_OPTIONS}
 
   progress "Analyzing netdata..."
-  run "${covbuild}" --dir cov-int make -j${cpus} || exit 1
+  run "${covbuild}" --dir cov-int make -j${cpus}
 
   echo >&2 "Compressing analysis..."
-  run tar czvf netdata-coverity-analysis.tgz cov-int || exit 1
+  run tar czvf netdata-coverity-analysis.tgz cov-int
 
   echo >&2 "Sending analysis to coverity for netdata version ${version} ..."
   COVERITY_SUBMIT_RESULT=$(debugrun curl --progress-bar \
     --form token="${token}" \
-    --form email=${email} \
+    --form email="${email}" \
     --form file=@netdata-coverity-analysis.tgz \
     --form version="${version}" \
     --form description="netdata, monitor everything, in real-time." \
-    https://scan.coverity.com/builds?project=${repo})
+    https://scan.coverity.com/builds?project="${repo}")
 
-  echo ${COVERITY_SUBMIT_RESULT} | grep -q -e 'Build successfully submitted' || echo >&2 "scan results were not pushed to coverity. Message was: ${COVERITY_SUBMIT_RESULT}"
+  echo "${COVERITY_SUBMIT_RESULT}" | grep -q -e 'Build successfully submitted' || echo >&2 "scan results were not pushed to coverity. Message was: ${COVERITY_SUBMIT_RESULT}"
 
   progress "Coverity scan completed"
 }
 
 installit() {
-  progress "Downloading coverity..."
-  cd /tmp || exit 1
+  ORIGINAL_DIR="${PWD}"
+  TMP_DIR="$(mktemp -d /tmp/netdata-coverity-scan-XXXXX)"
+  progress "Downloading coverity in ${TMP_DIR}..."
+  cd "${TMP_DIR}"
 
-  [ -f "${COVERITY_BUILD_VERSION}.tar.gz" ] && run rm -f "${COVERITY_BUILD_VERSION}.tar.gz"
   debugrun curl --remote-name --remote-header-name --show-error --location --data "token=${token}&project=${repo}" https://scan.coverity.com/download/linux64
 
   if [ -f "${COVERITY_BUILD_VERSION}.tar.gz" ]; then
     progress "Installing coverity..."
-    cd /opt || exit 1
-    run sudo tar -z -x -f  "/tmp/${COVERITY_BUILD_VERSION}.tar.gz" || exit 1
-    rm "/tmp/${COVERITY_BUILD_VERSION}.tar.gz"
-    export PATH=${PATH}:/opt/${COVERITY_BUILD_VERSION}/bin/
+    cd "${INSTALL_DIR}"
```

Comment:
why did you remove the `exit 1` guard?

Reply:
I 've added `set -e` at the start of the script that is basically stopping execution on any failed command, so no need to exit there

Reply:
ditto, `set -e` covers us